OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -68,7 +68,15 @@ static class DMOpcodeHandlers {
 
         public static ProcStatus? CreateObject(DMProcState state) {
             DreamProcArguments arguments = state.PopArguments();
-            DreamPath objectPath = state.PopDreamValue().GetValueAsPath();
+            var val = state.PopDreamValue();
+            if (!val.TryGetValueAsPath(out var objectPath) && val.TryGetValueAsString(out var pathString))
```

Comment:
This will end up with `objectPath` being `DreamPath.Root` if `val` is neither a path or a string, and will go on to create it. It should error instead.